java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/preparation/TransportTags.java

```diff
@@ -0,0 +1,111 @@
+package net.osmand.obf.preparation;
+
+import net.osmand.osm.edit.Relation;
+
+import java.util.*;
+
+public class TransportTags {
+	public static final int POPULARITY_THRESHOLD = 3;
+	private final Map<Long, List<TransportTagValue>> tags = new HashMap<>();
+	List<TransportTagValue> allTags = new ArrayList<>();
+	private static final Set<String> tagsFilter = new HashSet<>();
+
+	static {
+		tagsFilter.add("interval");
+		tagsFilter.add("opening_hours");
+		tagsFilter.add("duration");
+	}
+
+	public void add(Long routeId, List<TransportTagValue> transportTags) {
+		tags.put(routeId, transportTags);
+	}
+
+	public List<TransportTagValue> get(long idRoute) {
+		return tags.get(idRoute);
+	}
+
+	public void putFilteredTags(Relation rel, long routeId) {
+		Map<String, String> relTags = rel.getTags();
+		putFilteredTags(routeId, relTags);
+	}
+
+	public void putFilteredTags(long routeId, Map<String, String> relTags) {
+		List<TransportTagValue> filteredTags = new ArrayList<>();
+		for (String tagKey : relTags.keySet()) {
+			for (String neededTag : tagsFilter) {
+				if (tagKey.startsWith(neededTag)) {
+					TransportTagValue tagValue = new TransportTagValue(tagKey, relTags.get(tagKey));
```

Comment:
too slow and too cumbersome!
List<TransportTagValue> allTags = new ArrayList<>(); -> 
Map<String, TransportTagValue> allTags = // map key = (tag / value) ->